NOTICE.txt

```diff
@@ -1,2 +1,4 @@
-connectors-py
+Elastic Python connectors
 Copyright 2022 Elasticsearch B.V.
```

Comment:
Should this be
```suggestion
Copyright 2022-2023 Elasticsearch B.V.
```

Reply:
Probably just 2023?

Reply:
I didn't find any doc about this, but my understanding is it's a range from when the software was first developed to the current year. cc @elastic/open-source-wg